api/create.go

```diff
@@ -92,25 +92,16 @@ func CreateService(w http.ResponseWriter, req *http.Request) {
 	}
 	authHeader := req.Header.Get("Authorization")
 	apiKeyHeader := req.Header.Get("x-api-key")
-	if apiKeyHeader == "" {
-		userId, err := config.ValidateToken(authHeader)
-		if err != nil {
-			log.Printf("error validating token %v", err)
-			http.Error(w, "error validating token", 500)
-		}
-		if s.UserID != userId {
-			log.Printf("user %s trying to access /createservice using jwt userId %s", s.UserID, userId)
-			http.Error(w, "userid doesn't match", http.StatusInternalServerError)
-			return
-		}
+	userId, err := config.ValidateUser(authHeader, apiKeyHeader)
+
+	if err != nil {
+		http.Error(w, err.Error(), 500)
```

Comment:
Looks good to me.

Little nitpick though...since it's an error with validation/authentication, we could return a 401 instead - so maybe you could change the `500` here and the other parts to [`http.StatusUnauthorized`](https://pkg.go.dev/net/http#StatusUnauthorized)